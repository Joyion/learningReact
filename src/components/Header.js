import React from 'react';
import {NavLink} from 'react-router-dom';

const Header = () => (
    <div>
       <h2>Expense App</h2>
       <NavLink activeClassName="is-active"  exact={true} to="/">Home</NavLink>
       <NavLink activeClassName="is-active" to="/create">Create</NavLink>
       <NavLink activeClassName="is-active" to="/edit">Edit</NavLink>
    </div>
   
);

export default Header;