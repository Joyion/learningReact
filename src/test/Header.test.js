import ReactShallowRenderer from "react-test-renderer/shallow";
import React from "react";
import Header from "../components/Header"
import {shallow} from 'enzyme';





test("test header", () => {
    const wrapper = shallow(<Header />);
    expect(wrapper).toMatchSnapshot();
    
    // enzyme test
 //   expect(wrapper.find('h2').length).toBe(1);

    // const renderer = new ReactShallowRenderer();
    // renderer.render(<Header />);
    // expect(renderer.getRenderOutput()).toMatchSnapshot();
});