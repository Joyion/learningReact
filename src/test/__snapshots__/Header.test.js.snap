// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test header 1`] = `
<div>
  <h2>
    Expense App
  </h2>
  <NavLink
    activeClassName="is-active"
    exact={true}
    to="/"
  >
    Home
  </NavLink>
  <NavLink
    activeClassName="is-active"
    to="/create"
  >
    Create
  </NavLink>
  <NavLink
    activeClassName="is-active"
    to="/edit"
  >
    Edit
  </NavLink>
</div>
`;
